lib/rules/font-weight-notation/index.js

```diff
@@ -78,6 +79,8 @@ const rule = function(expectation, options) {
     }
 
     function checkWeight(weightValue, decl) {
+      const ignoreFunctions = (options && options.ignoreFunctions) || [];
```

Comment:
I think we'll want to use to the `optionsMatches` utility here, for consistency and because it supports regex.

I think we'll also want to ensure that we're only checking the function name. We can do that with the value parser. There's an example of that [here](https://github.com/stylelint/stylelint/blob/97d351e7246ae5acc7109cebdce17ae7ed8bdb30/lib/rules/unit-no-unknown/index.js#L44-L53) e.g.

```js
valueParser(weight).walk(function(valueNode) {
        if (
          valueNode.type === "function" &&
            optionsMatches(options, "ignoreFunctions", valueNode.value))
        ) {
          return false;
        }
});
```